### PR TITLE
Contributor guidelines

### PR DESCRIPTION
**Pull request by njgheorghita**
Just the beginning of a contributor guidelines section, a document outlining the various code hygiene and style tips that pr's should follow. The majority of this was lifted from the [snake charmers tactical manual](https://github.com/ethereum/snake-charmers-tactical-manual/). This is by no means a final version of these guidelines, and contributors are encouraged to update this document as the project grows and we recognize new guidelines that we want the project to follow.

### CODE REVIEW

File: contributor_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+want to contribute back to the PR (and the author has this enabled), we would
+still need to add their remote explicitly.
+
+## Comments
+
+- Any datatype of significance **should** have an accompanying comment briefly describing its role and responsibilities. Comments are an extremely valuable tool in open-source projects with many different contributors, and can greatly improve development speed. Explain your assumptions clearly so others don't need to dig through the code.
+
+
+## Error handling
+- Handle errors, avoid naked `.unwrap()`s.
```
Comment (mrferris) [inline contributor_guidelines.md:170]:
```suggestion
- Handle errors, avoid naked `.unwrap()`s except for in unit tests. Write descriptive error messages that give context of the problem that occured.
```

**Review by mrferris — COMMENTED** [review]
Looks good, only other section I would add as part of the initial version is a testing section. Here's something quick (and also stolen from Snake Charmers 🐍):

## Tests

Testing is essential to the production of software with minimal flaws. The default should always be writing tests for the code you produce. 

Testing also introduces overhead into our workflow. If a test suite takes a long time to run, it slows down our iteration cycle. This means finding a pragmatic balance between thorough testing, and the speed of our test suite, as well as always iterating on our testing infrastructure.

File: contributor_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
+# Contributor Guidelines
```
Comment (ogenev) [inline contributor_guidelines.md:1]: I would change this file name to CONTRIBUTING.md, I think it is more github idiomatic :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+We are a distributed team.  The primary way we communicate about our code is
+through github via pull requests.
+
+* When you start work on something you should have a pull request opened that
+  same day.
+* Mark unfinished pull requests with the "Work in Progress" label.
+* Before submitting a pr for review, you should run the following commands
+  locally and make sure they are passing, otherwise CI will raise an error.
+  * `cargo fmt` and `cargo clippy` for linting checks
```
Comment (carver) [inline CONTRIBUTING.md:25]: `cargo fmt -- --check` and `cargo clippy -- --deny warnings`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  * `cargo test --workspace` to run all tests
```
Comment (carver) [inline CONTRIBUTING.md:26]: `RUSTFLAGS='-D warnings' cargo test --workspace`

File: README.md [hunk 5]
```diff
@@ -134,12 +134,16 @@ OPTIONS:
...
+
 ## Gotchas
 
 - There is a limit on concurrent connections given by the threadpool. At last
   doc update, that number was 2, but will surely change. If you leave
   connections open, then new connections will block.
 - Error handling is pretty close to non-existent.
-- This project may never be updated. If this repo is looking stale, you might
-  try [asking the Trinity team](https://gitter.im/ethereum/trinity) to find out
-  what spiritual successor exists, if any.
```
Comment (carver) [inline README.md:145]: lol that was old

**Review by carver — COMMENTED** [review]
Thanks for putting this together, Nick!

**Pull request merged**